nikola/nikola.py

```diff
@@ -823,12 +823,19 @@ def __init__(self, **config):
                     utils.LOGGER.warn('You are currently disabling "{}", but not the following new taxonomy plugins: {}'.format(old_plugin_name, ', '.join(missing_plugins)))
                     utils.LOGGER.warn('Please also disable these new plugins or remove "{}" from the DISABLED_PLUGINS list.'.format(old_plugin_name))
                     self.config['DISABLED_PLUGINS'].extend(missing_plugins)
+        # Special-case logic for "render_indexes" to fix #2591
+        if 'render_indexes' in self.config['DISABLED_PLUGINS']:
+            if 'generate_rss' in self.config['DISABLED_PLUGINS'] or self.config['GENERATE_RSS'] is False:
+                self.config['DISABLED_PLUGINS'].append('classify_indexes')
+            else:
+                self.config['DISABLE_INDEXES_PLUGIN_INDEX_AND_ATOM_FEED'] = True
 
         # Disable RSS.  For a successful disable, we must have both the option
         # false and the plugin disabled through the official means.
         if 'generate_rss' in self.config['DISABLED_PLUGINS'] and self.config['GENERATE_RSS'] is True:
             utils.LOGGER.warn('Please use GENERATE_RSS to disable RSS feed generation, instead of mentioning generate_rss in DISABLED_PLUGINS.')
             self.config['GENERATE_RSS'] = False
+            self.config['DISABLE_INDEXES_PLUGIN_RSS_FEED'] = True
```

Comment:
Sure. The messing with disabling plugins is so that backwards compatibility should be kept. I'm adding some more warning messages.